runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -112,13 +114,97 @@ func (q *MetricsViewComparison) Resolve(ctx context.Context, rt *runtime.Runtime
 		return err
 	}
 
+	// comparison toplist
 	if !isTimeRangeNil(q.ComparisonTimeRange) {
+		// execute toplist for base and get dim list
+		// create and add filter and execute comprison toplist
+		// remove strict limits in comp toplist sql
+		if drivers.DialectDruid != olap.Dialect() || q.Exact {
```

Comment:
I'm worried about how inlined/nested/scattered this fix is since the comparison logic is already tricky to follow. I'm wondering if it could we moved into a separate branch of logic without altering the existing implementation too much? For example, something like this on line 117:

```go
if olap.Dialect() == drivers.DialectDruid && !q.Exact {
  return q.executeDruidApproximate(ctx, olap, mv, priority, security)
}
```